src/Control/Effect/Reader.hs

```diff
@@ -1,4 +1,16 @@
 {-# LANGUAGE DeriveFunctor, ExistentialQuantification, FlexibleContexts, StandaloneDeriving #-}
+
+{- | An effect providing access to an immutable (but locally-modifiable) context value.
+
+This effect is similar to the traditional @MonadReader@ typeclass, though it allows the
+presence of multiple @Reader t@ effects, as long as the values of @t@ are distinct.
```

Comment:
1. Soft wrap, please 🙏
2. There’s no requirement that they be distinct, it’s just that it won’t be able to distinguish requests for multiple identical effects (as with any other effect!) and the requests will go to the leftmost.